qute.ls/com.redhat.qute.ls/src/main/java/com/redhat/qute/services/QuteCodeActions.java

```diff
@@ -298,4 +305,32 @@ private static void doCodeActionsForUndefinedSectionTag(Template template, Diagn
 			LOGGER.log(Level.SEVERE, "Creation of undefined user tag code action failed", e);
 		}
 	}
+
+	private static void doCodeActionsForSimilarValues(Template template, Diagnostic diagnostic,
+			List<CodeAction> codeActions, String varName) throws BadLocationException {
+		int offset = template.offsetAt(diagnostic.getRange().getStart());
+		Node node = QutePositionUtility.findBestNode(offset, template.findNodeAt(offset));
+
+		Range diagnosticRange = diagnostic.getRange();
+
+		Range rangeValue = new Range(
+				new Position(diagnosticRange.getStart().getLine(), diagnosticRange.getStart().getCharacter() + 1),
+				new Position(diagnosticRange.getEnd().getLine(), diagnosticRange.getEnd().getCharacter() - 1));
+		List<String> tagCompletions = Arrays.asList("for", "if", "each");

```

Comment:
I assume it is just to initialize the "Did you mean" code action that you hard code the list. The issue is for expression not for tag section (we could do that if you wish in a sperate PR).

Once you get the node,  you need to tests if it is a part, and according to part.getKind() you will need to fill the tagCompletions (which should be renamed with availableValues). See https://github.com/redhat-developer/quarkus-ls/blob/fc083c8d7e458b7f3222d6f63c381662a9a41c77/qute.ls/com.redhat.qute.ls/src/main/java/com/redhat/qute/services/QuteHover.java#L183

I suggest that you see this hover code, for instance for getting values of an objct part, you should see https://github.com/redhat-developer/quarkus-ls/blob/fc083c8d7e458b7f3222d6f63c381662a9a41c77/qute.ls/com.redhat.qute.ls/src/main/java/com/redhat/qute/services/QuteHover.java#L333

ResolvedJavaType javaType = javaCache.resolveJavaType(previousPart, projectUri).getNow(null);
if (javaType != null) {
   javaType.getMethods()....
}